### PR TITLE
Update docs to use Sphinx with RST and RTD theme

### PR DESCRIPTION
**Pull request by rdkr**
This PR combines the recent work @1oglop1 and I have done to update the docs to use reStructuredText to generate HTML using Sphinx and the RTD theme. This cleanly brings together the end user and API docs, makes the docs clearer to navigate, and allows use of RST features for doc authors.

A copy of the docs generated from the incoming branch are [temporarily viewable here](http://neel-sceptre.s3-website.eu-west-2.amazonaws.com/index.html).

This PR also relates to:
* #306 (Update Documentation for v2.0)
* #389 (Breakout docs in to their own repository)

I've done my best to clean up the commits whilst maintaining authorship, given that this work was done by two people.

-----------------

* [x] Wrote [good commit messages][1].
* [x] [Squashed related commits together][2] after the changes have been approved.
* [x] Commit message starts with `[Resolve #issue-number]` (if a related issue exists).
* ~[ ] Added unit tests.~
* ~[ ] Added integration tests (if applicable).~
* ~[ ] All unit tests (`make test`) are passing.~
* ~[ ] Used the same coding conventions as the rest of the project.~
* [x] The new code doesn't generate flake8 (`make lint`) offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

### CODE REVIEW

**Comment by ngfgrant** [comment]
Hi guys - just to let you know I am looking at this PR this week, feedback in due course.

File: docs/_source/index.rst [hunk 1]
```diff
@@ -33,19 +28,14 @@ and corresponding YAML config files. The config files include which
...
 Code
 ----
 
-Sceptre’s source code can be found on `Github`_.
-
-Bugs and feature requests should be raised via our `Issues`_ page.
-
-This documentation acts as an in-depth reference for all of Sceptre's features.
-
-Beginners should consider reading our :doc:`docs/get_started` ..., which provides a walk-through tutorial.
```
Comment (ngfgrant) [inline docs/_source/index.rst]: Why remove the link towards the getting started guide?

File: docs/_source/conf.py [hunk 2]
```diff
@@ -164,9 +164,15 @@
...
-    (master_doc, 'Sceptre', 'Sceptre Documentation',
-     author, 'Sceptre', 'One line description of project.',
-     'Miscellaneous'),
+    (
+        master_doc,
+        "Sceptre",
+        "Sceptre Documentation",
+        author,
+        "Sceptre",
+        "One line description of project.",
```
Comment (ngfgrant) [inline docs/_source/conf.py]: Although this line was not introduced here may as well fix it. "Automate cloud deployments" is the tag line on the home page so lets go with that.

File: docs/_source/conf.py [hunk 3]
```diff
@@ -164,9 +164,15 @@
...
 #  dir menu entry, description, category)
 texinfo_documents = [
-    (master_doc, 'Sceptre', 'Sceptre Documentation',
-     author, 'Sceptre', 'One line description of project.',
-     'Miscellaneous'),
+    (
+        master_doc,
+        "Sceptre",
+        "Sceptre Documentation",
+        author,
```
Comment (ngfgrant) [inline docs/_source/conf.py]: author -> Cloudreach

File: docs/_source/conf.py [hunk 4]
```diff
@@ -20,18 +20,20 @@
...
+project_path = os.path.join(
+    os.path.dirname(__file__), ".." + os.path.sep + ".."
+)
 sys.path.insert(0, os.path.abspath(project_path))
 # -- Project information -----------------------------------------------------
 
-project = 'Sceptre'
-copyright = '2018, Cloudreach'
-author = 'Cloudreach'
+project = "Sceptre"
```
Comment (ngfgrant) [inline docs/_source/conf.py]: The default and convention in the conf.py file is to use single quotes, can you please revert the changes that introduced the double quotes. Looks like a editor probably changed these automatically.

File: docs/_source/conf.py [hunk 5]
```diff
@@ -20,18 +20,20 @@
 import sys
 
 # Get the project root dir, which is the parent dir of this
-project_path = os.path.join(os.path.dirname(__file__), '..' + os.path.sep + '..')
+project_path = os.path.join(
+    os.path.dirname(__file__), ".." + os.path.sep + ".."
```
Comment (ngfgrant) [inline docs/_source/conf.py]: No real benefit breaking this away from a single line, I doubt it was breaking a lint?. If you want to "improve" this one way to that is to do something like:

```python
from os.path import join
from os.path import dirname


project_path = join(dirname(__file__), join("..", ".."))
```

Otherwise, the original was fine as is.

File: docs/_source/docs/resolvers.rst [hunk 6]
```diff
@@ -187,7 +187,7 @@ setup.py
     setup(
         name='custom_resolver',
         entry_points={
-            'sceptre.hooks': [
+            'sceptre.resolvers': [
```
Comment (ngfgrant) [inline docs/_source/docs/resolvers.rst]: When you rebase you will find this conflicts with the latest docs. It might be easier when you rebase to drop this change.

File: docs/_source/index.rst [hunk 7]
```diff
@@ -70,6 +59,7 @@ Beginners should consider reading our :doc:`docs/get_started` ..., which provide
    docs/templates.rst
    docs/hooks.rst
    docs/resolvers.rst
+   docs/architecture.rst
```
Comment (ngfgrant) [inline docs/_source/index.rst]: Architecture made more sense in the previous position so can we keep it there?

File: requirements.txt [hunk 8]
```diff
@@ -20,6 +20,7 @@ setuptools==37.0.0
 six==1.11.0
 Sphinx>=1.6.5
 sphinx_rtd_theme>=0.4.0
+sphinx-click>=1.2.0
```
Comment (ngfgrant) [inline requirements.txt]: `sphinx-click>=1.2.0,<2.0`

File: docs/_source/docs/cli.rst [hunk 9]
```diff
@@ -6,64 +6,11 @@ format:
 
 .. code:: shell
 
-    $ sceptre [GLOBAL_OPTIONS] COMMAND [ARGS] [COMMAND_OPTIONS]
+    $ sceptre
```
Comment (ngfgrant) [inline docs/_source/docs/cli.rst]: Can you please revert this line back to include `sceptre [GLOBAL_OPTIONS] COMMAND [ARGS] [COMMAND_OPTIONS]`? Was there a reason you just kept `$sceptre` here?

File: requirements.txt [hunk 10]
```diff
@@ -18,7 +18,8 @@ PyYaml==3.12
 readme-renderer==17.2
 setuptools==37.0.0
 six==1.11.0
-Sphinx==1.6.5
+Sphinx>=1.6.5
```
Comment (ngfgrant) [inline requirements.txt]: `Sphinx>=1.6.5, <2.0`

File: requirements.txt [hunk 11]
```diff
@@ -18,7 +18,8 @@ PyYaml==3.12
...
+sphinx_rtd_theme>=0.4.0
```
Comment (ngfgrant) [inline requirements.txt]: pre v1 release so please change to `sphinx_rtd_theme==0.4.0`

File: docs/_source/conf.py [hunk 12]
```diff
@@ -100,7 +100,7 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+# html_static_path = ['_static']
```
Comment (ngfgrant) [inline docs/_source/conf.py]: Just want to check this was deliberate?

**Comment by rdkr** [comment]
Closing as superseded by https://github.com/cloudreach/sceptre/pull/567
